### DISCUSSION TITLE
4k Spout / GL texture over network?

### DISCUSSION THREAD

**Discussion by antithing** [CESNET/UltraGrid#230] [Q&A]
Hi, and thank you for making this code available. 

My use case is the following:

Capture 4 x  GLuint textures simultaneously using Spout
Send each to a different machine over a network.
Receive each, and stream via spout to a third party application on each machine.

Can this be done with this framework?

Is there any example code that does a similar thing?

Thank you.

**Comment by MartinPulec** [comment]
Hi @antithing,

if I understand the setup correctly, it is quite easy - the source will be one machine (sender) and 4 receivers (recv1-4), correct? If so, then:

    sender$ uv -t spout:name=texture1 -c <compression> recv1
    sender$ uv -t spout:name=texture2 -c <compression> recv2
    sender$ uv -t spout:name=texture3 -c <compression> recv3
    sender$ uv -t spout:name=texture4 -c <compression> recv4

    recv1-recv4$ uv -d gl:spout:hide_window

The above is just a kick-off, I didn't intentionally specify the compression since it hugely depends on the actual scenario - if it is 4x 4K picture, you'd perhaps need some HW-accelerated codec unless only a small frame-rate is requested.

**Comment by antithing** [comment]
Thank you! Very helpful. I'll try this.
